mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -137,37 +174,47 @@ const GraphSidebar = (props: {
     */
     const [graphUpdatedNumber, setGraphUpdatedNumber] = useState(0)
 
-    const dataSourceSheetIndex = graphParams.graphCreation.sheet_index
-    const graphOutput = props.graphDataJSON[dataSourceSheetIndex]?.graphOutput
-    const [_copyGraphCode, graphCodeCopied] = useCopyToClipboard(graphOutput?.graphGeneratedCode);
-    const [stepID, setStepID] = useState<string|undefined>(undefined);
-
-    const [loading, setLoading] = useState<boolean>(false)
-
     // Save the last step index, so that we can check if an undo occured
     const prevLastStepIndex = usePrevious(props.lastStepIndex);
+    // Save the last number of graphs so that we can check if a graph was deleted during an undo, 
+    // so we can update the selected tab
+    const prevNumberOfGraphs = usePrevious(Object.keys(props.graphDataJSON).length)
 
     // When the last step index changes, check if an undo occured so we can refresh the params
     useEffect(() => {
         // If there has been an undo, then we refresh the params to this pivot
         if (prevLastStepIndex && prevLastStepIndex !== props.lastStepIndex - 1) {
             void refreshParamsAfterUndo()
+
+            const currNumGraphs = Object.keys(props.graphDataJSON).length
+            // If the number of graphs changed, update the selected tab
+            if (prevNumberOfGraphs && prevNumberOfGraphs > currNumGraphs) {
+                selectPreviousGraphSheetTab(props.graphDataJSON, props.setUIState)
```

Comment:
Can we rename this function to `selectRightmostGraphSheetTab`?

Reply:
Ignoring since we changed the functionality away from selecting the rightmost graph sheet tab.